coriolis/tests/api/v1/test_migrations.py

```diff
@@ -0,0 +1,163 @@
+                    "instances": ["mock_instance1", "mock_instance2"],
+                    "replica_id": "mock_replica_id",
+                    "clone_disks": True,
+                    "force": False,
+                    "skip_os_morphing": False,
+                    "instance_osmorphing_minion_pool_mappings":
+                        {"mock_mapping": "mock_value"},
+                }
+            }
+        ),
+        (
+            {
+                "expected_api_method": 'migrate_instances',
+                "validation_expected": True,
+                "migration": {
+                    "user_scripts": {"mock_user_scripts": None},
+                    "instances": ["mock_instance1", "mock_instance2"],
+                    "replica_id": None,
+                    "clone_disks": True,
+                    "force": False,
+                    "skip_os_morphing": False,
+                    "instance_osmorphing_minion_pool_mappings":
+                        {"mock_mapping": "mock_value"},
+                }
+            }
+        )
+    )
+    @mock.patch.object(migration_view, 'single')
+    @mock.patch.object(migrations.MigrationController,
+                       '_validate_migration_input')
+    @mock.patch.object(api_utils, 'normalize_user_scripts')
+    @mock.patch.object(api_utils, 'validate_user_scripts')
+    @mock.patch.object(migration_policies, 'get_migrations_policy_label')
+    def test_create(
+        self,
+        data,
+        mock_get_migrations_policy_label,
+        mock_validate_user_scripts,
+        mock_normalize_user_scripts,
+        mock__validate_migration_input,
+        mock_single
+    ):
+        expected_api_method = data.pop('expected_api_method')
+        validation_expected = data.pop('validation_expected', False)
+        with mock.patch.object(api.API,
+                               expected_api_method) as mock_api_method:
+            mock_req = mock.Mock()
+            mock_context = mock.Mock()
+            mock_req.environ = {'coriolis.context': mock_context}
+            mock__validate_migration_input.return_value = \
+                (mock.sentinel.value,) * 14
+
+            result = self.migrations.create(mock_req, data)
+
+            self.assertEqual(
+                mock_single.return_value,
+                result
+            )
+
+            mock_get_migrations_policy_label.assert_called_with("create")
+            mock_validate_user_scripts.assert_called_with(
+                data['migration']['user_scripts'])
+            mock_normalize_user_scripts.assert_called_with(
+                data['migration']['user_scripts'],
+                data['migration']['instances']
+            )
+            if validation_expected:
+                mock__validate_migration_input.assert_called_with(
+                    mock_context, data)
+            mock_api_method.assert_called_once()
+            mock_single.assert_called_with(mock_api_method.return_value)
```

Comment:
This is otherwise good so far, please add tests for the rest of the methods of `coriolis.api.v1.migrations`